File: C++/archivos cpp/count_smileys.cpp
#include<iostream>
#include<vector>
#include<regex>
using namespace std;

int count_smileys(vector<string> faces){
	int faces_found = 0;
	regex pattern("^[:;][-~]?[)D]$");
	for(auto& face : faces){
		(regex_match(face, pattern)) ? faces_found++ : faces_found;
	}
	return faces_found;
}

int main(int argc, char const *argv[]){
	cout<<count_smileys({";D", ":-(", ";~)", ":)", ";(", ";}", ":-D"});
	return 0;
}

File: Python/Práctica/count_smileys.py
import re

def count_smileys(faces):
	faces_found = 0
	for face in faces:
		if re.search("^[:;][-~]?[)D]$", face):
			faces_found+=1

	return faces_found

print(count_smileys([";D", ":-(", ";~)", ":)", ";(", ";}", ":-D"]))



